src/monitors/ControlInstrumenter.v3

```diff
@@ -72,18 +72,19 @@ class ControlInstrumentation(func: FuncDecl, entries: Array<(int, Probe)>) {
 		return if (e.0 == 0, CountProbe.!(e.1));
 	}
 	def getCount(pc: int) -> CountProbe {
-		// TODO: binary search
-		for (i < entries.length) {
+		var lower = RangeUtil.lowerbound(entries, (pc, null), cmp_lt_pc);
+		var upper = RangeUtil.upperbound(entries, (pc, null), cmp_lt_pc);
+		for (i = lower; i < upper; i++) {
 			var e = entries[i];
-			if (e.0 == pc && CountProbe.?(e.1)) return CountProbe.!(e.1);
+			if (CountProbe.?(e.1)) return CountProbe.!(e.1);
 		}
 		return null;
 	}
 	def getBrCounts(pc: int) -> Array<u64> {
-		// TODO: binary search
-		for (i < entries.length) {
-			var e = entries[i];
-			if (e.0 == pc) match (e.1) {
+		var lower = RangeUtil.lowerbound(entries, (pc, null), cmp_lt_pc);
```

Comment:
If you're looking for an exact match, I think it makes sense to have a utility method that specifically finds an exact match.